include/isc_nav/utility/rrt.hpp

```diff
@@ -0,0 +1,267 @@
+{
+    class RRT 
+    {
+    public:
+        explicit RRT(const nav_msgs::msg::OccupancyGrid& grid) : start_{}, goal_{}, map_(grid) {}
+
+        const nav_msgs::msg::Path get_path(const geometry_msgs::msg::Pose& start_pose, const geometry_msgs::msg::Pose& goal_pose)
+        {
+            nav_msgs::msg::Path path;
+            start_ = Point2D(start_pose.position.x, start_pose.position.y);
+            goal_ = Point2D(goal_pose.position.x, goal_pose.position.y);
+
+            Point2D random, Xnew, Xnearest;
+            double Xnearest_dist = 0;
+
+            int lim = 1000; // number of iterations
+
+            // Graph with vertices and edges
+            V.push_back(start_);
+            E.push_back(0);
+            G.push_back(E);
+
+            for (int i = 0; i < lim; i++)
+            {   
+                random = randomPosition();
+
+                // Find nearest vertex to the random node
+                Xnearest = findNearest(V, random, Xnearest_dist);
+
+                if (getDistance(Xnearest, random) <= max_dist_)
+                {
+                    // If random vertex is short
+                    Xnew = Xnearest;
+                }
+                else
+                {
+                    Xnew = findNew(Xnearest, random, Xnearest_dist);
+                }
+
+                if (isInObstacle(Xnew))
+                {
+                    continue;
+                }
+                
+                // Make sure no obstacle between Xnearest and Xnew
+                if (!isCollisionFree(Xnearest, Xnew))
+                {
+                    continue;
+                }
+
+                // Get the index of to and from
+                int attach_to = getIndex(V, Xnew);
+                int attach_from = getIndex(V, Xnearest);
+
+                // Add edge
+                if (attach_to == -1)
+                {
+                    int add_new = V.size();
+                    V.push_back(Xnew);
+                    E.push_back(0);
+
+                    for (int i = 0; i < attach_to; i++)
+                    {
+                        G.at(i).push_back(0);
+                    }
+
+                    G.at(attach_from).at(add_new)++;
+                    G.push_back(E);
+                }
+                else 
+                {
+                    V.push_back(Xnew);
+                    E.push_back(0);
+
+                    for (int i = 0; i < attach_to; i++)
+                    {
+                        G.at(i).push_back(0);
+                    }
+
+                    G.at(attach_from).at(attach_to)++;
+                    G.push_back(E);
+                }
+
+                // If vertex at goal return graph
+                if ((Xnew.x - goal_.x < 0.1) && (Xnew.y - goal_.y < 0.1))
+                {
+                    trace_back_path(Xnew, path);
+                }
+            }
+
+            return path;
+        }
+
+
+        Point2D randomPosition()
+        {
+            srand(time(0));
```

Comment:
I would advise against using c rand, for a lot of reasons. The way I would do RNG in C++ is like the following:
```cpp
  //do these 3 statements only once in a constructor
  std::random_device dev;
  std::mt19937 rng(dev()); //Store as a feild
  std::uniform_int_distribution<std::mt19937::result_type> dist6(1,6); // distribution in range [1, 6]. Store in a feild

  std::cout << dist6(rng) << std::endl; //Getting a random value
```

This has the advantage of not only speed, but also avoiding the pitfalls of a C API. For example, this line is re-seeding the rng every single call, which is a very bad idea.